airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -224,8 +236,8 @@ def read_records(
                 params=self.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
                 json=self.request_body_json(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
             )
-
-            response = self._send_request(request)
+            request_kwargs = self.request_kwargs(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
```

Comment:
not sure I understand why you split kwargs here? can't you pass it directly to PreparedRequest?

Reply:
ok, I see you tried to mimic `Session.request()` logic, what is the reason to use PreparedRequest?
is it just to not call `request_body_json` and other functions multiple times in case of retry? what if it is user wants? i.e. what if request fails because of one of its parameters that needs to be changed?

Reply:
using `Session.prepare_request` instead of `request.prepare()` or just `Session.request()`, has benefit of sharing session states like cookies across requests.

Reply:
we need to use prepared request for `Session.send`

good idea, changed to use `Session.prepare_request`